app/api/public/users/[fid]/status/route.ts

```diff
@@ -0,0 +1,112 @@
+import { NextRequest, NextResponse } from "next/server";
+import prisma from "../../../../../../lib/prisma";
+
+export const GET = async (
+  req: NextRequest,
+  {
+    params: { fid },
+  }: {
+    params: { fid: string };
+  }
+) => {
+  try {
+    // Get the API key from the headers
+    const apiKey = req.headers.get("x-api-key");
```

Comment:
Let's create a function that:
- gets in a req: NextRequest and mode: string parameters
- performs all the checks that you are doing between line 14 and line 44
- returns a boolean
- before returning, if the boolean result is true (api-key and mode are valid for the current endpoint), create a request in the db as you are doing on line 47